### PR TITLE
fix: [Common] Fix Coverity issues

### PR DESCRIPTION
**Pull request by tsaikevin**
Fixed dead default in switch and logically dead code (CWE 561)

### CODE REVIEW

File: BootloaderCommonPkg/Library/FatLib/FatLib.c [hunk 1]
```diff
@@ -185,7 +185,7 @@ FindFile (
           }
         }
       } while (Status == EFI_SUCCESS);
-      if (EFI_ERROR (Status)) {
```
Comment (bejeanmo) [inline BootloaderCommonPkg/Library/FatLib/FatLib.c:188]: Is this code actually a problem? It seems like coverity doesn't understand the INTN cast:
image
Seems like this should be traiged as a false positive in coverity instead.

Comment (tsaikevin) [reply BootloaderCommonPkg/Library/FatLib/FatLib.c]: @bejeanmo reverted changes, pls review again. thanks

**Review by bejeanmo — APPROVED** [review]

**Review by bejeanmo — CHANGES_REQUESTED** [review]
Update copyright dates please.

File: BootloaderCommonPkg/Library/XhciLib/XhcPeim.c [hunk 2]
```diff
@@ -917,9 +917,7 @@ XhcPeiBulkTransfer (
       return EFI_INVALID_PARAMETER;
     }
   } else {
-    if (((DeviceSpeed == EFI_USB_SPEED_LOW) && (MaximumPacketLength > 8))  ||
-        ((DeviceSpeed == EFI_USB_SPEED_FULL) && (MaximumPacketLength > 64)) ||
-        ((DeviceSpeed == EFI_USB_SPEED_HIGH) && (MaximumPacketLength > 3072))) {
+    if ((DeviceSpeed == EFI_USB_SPEED_LOW) && (MaximumPacketLength > 8)) {
```
[the remark this answers is not in the document]
Comment (tsaikevin) [reply BootloaderCommonPkg/Library/XhciLib/XhcPeim.c:920]: @bejeanmo the line 913 will take effect only in the case if (!IsInterruptTransfer), I guess it is still needed here. pls check again. thanks

[the remark this answers is not in the document]
Comment (tsaikevin) [reply BootloaderCommonPkg/Library/XhciLib/XhcPeim.c:919]: @bejeanmo pls review again.

**Comment by tsaikevin** [comment]
> Update copyright dates please.

@bejeanmo update dates, pls review again. thanks

**Review by bejeanmo — APPROVED** [review]

**Pull request merged**